go-controller/pkg/util/pod_annotation.go

```diff
@@ -570,6 +590,16 @@ func AddRoutesGatewayIP(
 					podAnnotation.Gateways = append(podAnnotation.Gateways, gatewayIPnet.IP)
 				}
 			}
+			// Until https://github.com/ovn-kubernetes/ovn-kubernetes/issues/4876 is fixed, it is limited to IC only
+			if config.OVNKubernetesFeature.EnableInterconnect {
+				if _, isIPv6Mode := netinfo.IPMode(); isIPv6Mode {
+					joinAddrs, err := ParseNodeGatewayRouterJoinAddrs(node, netinfo.GetNetworkName())
+					if err != nil {
+						return err
+					}
+					podAnnotation.IPv6LLAGateway = HWAddrToIPv6LLA(IPAddrToHWAddr(joinAddrs[0].IP))
```

Comment:
shouldn't we check the `joinAddrs` are not empty ? 
Do you have a test for that scenario ?

Reply:
maybe already verified here  https://github.com/ovn-kubernetes/ovn-kubernetes/blob/6c366bb49d6fa57c80d6b39834952ce1d4d9dbb6/go-controller/pkg/ovn/secondary_layer2_network_controller.go#L622?

there are few unit tests about `ParseNodeGatewayRouterJoinAddrs`
i.e https://github.com/ovn-kubernetes/ovn-kubernetes/blob/6c366bb49d6fa57c80d6b39834952ce1d4d9dbb6/go-controller/pkg/clustermanager/clustermanager_test.go#L918
if related

Imho even if we should at something in the end (which might not be the case, if the first link is good enough), it should be minimal, and not over testing,

Reply:
I don't see the relationship between your code and `addPortForRemoteNodeGR` (the fn you've linked).

IMHO it would have been OK if the parsing function checked it, but, we see it doesn't.

Reply:
would it be fine to just add an assert here ?
UT sounds redundant

Reply:
added a condition, please let me know if you request beside